File: conferenti-api/src/Conferenti.Domain/Sessions/Session.cs
using System.Text.Json.Serialization;

namespace Conferenti.Domain.Sessions;

/// <summary>
/// Detailed information about a session given by a speaker
/// </summary>
public class Session
{
    /// <summary>
    /// The Session id used to get session details
    /// </summary>
    [JsonPropertyName("sessionId")]
    public string SessionId { get; set; }

    /// <summary>
    /// Title of the session
    /// </summary>
    [JsonPropertyName("title")]
    public string Title { get; set; }

    /// <summary>
    /// Slug of the session
    /// </summary>
    [JsonPropertyName("slug")]
    public string Slug { get; set; }

    /// <summary>
    /// Tags associated with the session
    /// </summary>
    [JsonPropertyName("tags")]
    public string[] Tags { get; set; }

    /// <summary>
    /// Description of the session
    /// </summary>
    [JsonPropertyName("description")]
    public string Description { get; set; }

    /// <summary>
    /// Start time of the session
    /// </summary>
    [JsonPropertyName("startTime")]
    public DateTime StartTime { get; set; }

    /// <summary>
    /// End time of the session
    /// </summary>
    [JsonPropertyName("endTime")]
    public DateTime EndTime { get; set; }

    /// <summary>
    /// Room where the session is held
    /// </summary>
    [JsonPropertyName("room")]
    public string Room { get; set; }

    /// <summary>
    /// Level of the session (e.g., Beginner, Intermediate, Advanced)
    /// </summary>
    [JsonPropertyName("level")]
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public SessionLevel Level { get; set; }

    /// <summary>
    /// Format of the session (e.g., Workshop, Lecture, Panel)
    /// </summary>
    [JsonPropertyName("format")]
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public SessionFormat Format { get; set; }

    /// <summary>
    /// Language in which the session is conducted
    /// </summary>
    [JsonPropertyName("language")]
    public string Language { get; set; }

    /// <summary>
    /// The IDs of the speakers presenting the session
    /// </summary>
    [JsonPropertyName("speakerIds")]
    public string[] SpeakerIds { get; set; }
}

File: conferenti-ai-agent/src/conferenti_agent/types/session.py
from dataclasses import dataclass
from datetime import datetime
from enum import Enum
from typing import List


class SessionLevel(str, Enum):
    BEGINNER = "Beginner"
    INTERMEDIATE = "Intermediate"
    ADVANCED = "Advanced"


class SessionFormat(str, Enum):
    LECTURE = "Lecture"
    WORKSHOP = "Workshop"
    PANEL = "Panel"
    KEYNOTE = "Keynote"
    PRESENTATION = "Presentation"


@dataclass
class Session:
    id: str
    title: str
    slug: str
    tags: List[str]
    description: str
    startTime: datetime
    endTime: datetime
    room: str
    level: SessionLevel
    format: SessionFormat
    language: str
    speakerIds: List[str]

    def to_cosmos_dict(self) -> dict:
        return {
            "id": self.id,
            "title": self.title,
            "slug": self.slug,
            "tags": self.tags,
            "description": self.description,
            "startTime": self.startTime.isoformat(),
            "endTime": self.endTime.isoformat(),
            "room": self.room,
            "level": self.level.value,
            "format": self.format.value,
            "language": self.language,
            "speakerIds": self.speakerIds,
        }

    @classmethod
    def from_cosmos_dict(cls, data: dict) -> "Session":
        return cls(
            id=data["id"],
            title=data["title"],
            slug=data["slug"],
            tags=data["tags"],
            description=data["description"],
            startTime=datetime.fromisoformat(data["startTime"]),
            endTime=datetime.fromisoformat(data["endTime"]),
            room=data["room"],
            level=SessionLevel(data["level"]),
            format=SessionFormat(data["format"]),
            language=data["language"],
            speakerIds=data["speakerIds"],
        )


@dataclass
class SessionCreateRequest:
    """Request model for creating sessions via AI"""

    id: str
    session_name: str
    target_audience: str
    industry: str
    focus_areas: List[str]
    audience_level: SessionLevel
    number_of_sessions: int = 10
    session_duration: int = 45  # minutes
    language: str = "English"
